$OpenBSD: patch-git-svn_perl,v 1.1 2009/01/29 23:16:35 bernd Exp $

In svn 1.5.5 the output of "svn info" for added directories was changed
and now shows the repository UUID. This patch implements the same
behavior for "git svn info" and makes t9119-git-svn-info.17 pass if
svn 1.5.5 is used.

Patch from Marcel Koeppen, found on the git mailing list.

--- git-svn.perl.orig	Thu Jan 29 10:43:46 2009
+++ git-svn.perl	Thu Jan 29 20:13:00 2009
@@ -911,7 +911,8 @@ sub cmd_info {
 	if ($@) {
 		$result .= "Repository Root: (offline)\n";
 	}
-	$result .= "Repository UUID: $uuid\n" unless $diff_status eq "A";
+	$result .= "Repository UUID: $uuid\n" unless $diff_status eq "A" &&
+		($SVN::Core::VERSION le '1.5.4' || $file_type ne "dir");
 	$result .= "Revision: " . ($diff_status eq "A" ? 0 : $rev) . "\n";
 
 	$result .= "Node Kind: " .
